nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -1195,14 +1196,14 @@ impl DataStore {
             })
     }
 
-    /// Attempt to read the latest collection while limiting queries to `limit`
-    /// records
+    /// Attempt to read the latest collection while limiting queries to `batch_size`
+    /// records at a time.
     ///
     /// If there aren't any collections, return `Ok(None)`.
     pub async fn inventory_get_latest_collection(
         &self,
         opctx: &OpContext,
-        limit: NonZeroU32,
+        batch_size: NonZeroU32,
```

Comment:
I suspect it's not important to allow consumers to control the batch size.  I think I'd probably just define a `const` in this file of like 500 or 1000 and use that here.  (You'd get to remove one or two apologetic block comments associated with arbitrary constants like [this one](https://github.com/oxidecomputer/omicron/blob/main/nexus/src/app/rack.rs#L74-L85)!).  If there's a consumer that needs to control it, maybe we could accept `Option<NonZeroU32>` and default to a `const` in this file?  Basically I don't want to have N consumers have to make this decision when it really shouldn't matter for most of them.

Update: I just found [this example](https://github.com/oxidecomputer/omicron/blob/5ba928fb9b661fb46142be8c63552e9e7d7c6278/nexus/src/app/background/sync_service_zone_nat.rs#L65), which is definitely dangerous, and so I'm more convinced we *don't* want callers to have to provide something here.